### DISCUSSION TITLE
Gradio/Default UI seems to be broken

### DISCUSSION THREAD

**Discussion by Rex109** [vladmandic/sdnext#2518] [Q&A]
Hey everyone is there any way to restore the old ui? This one seems to be stretching all the way down like a mobile view and it's really uncomfortable to use, everytime i have to scroll down to modify parameters and see my generated image.

![image](https://github.com/vladmandic/automatic/assets/8615649/b54177f0-65d8-4dae-8dac-628c8449d22b)

**Comment by vladmandic** [comment]
that is NOT new ui - that is YOUR choice of some theme, most likely gradio default - which is like name says - default by gradio, not optimized in any form for sdnext. use any of the primary built-in themes.

**Reply by Rex109** [reply]
Yeah but it was working fine before, is there any similar alternative that i can use? (this is gradio/default)

**Reply by vladmandic** [reply]
it was working before because it was not really gradio/default - it was using mashup of multiple loaded css and it was causing quite a lot of issues.
if you want to use gradio or 3rd party themes, then sdnext style is no longer loaded - and that is what happens now. so if you don't like it, sorry, feel free to edit/improve/style anyway you want - but i'm not in business of supporting 3rd party themes.
built-in themes are themes that are natively supported.

and your statement "new ui looks aweful" is going to be misinterpreted by 99% of users. so be clear what you write. i'm not about to censor you, but when i see a title like that i'm anything but motivated to help.

**Reply by Rex109** [reply]
I apologize if my thread title was too aggressive, i've changed it. Hope it's ok now. I was just confused of why before it was working and now is completely changed.

**Reply by vladmandic** [reply]
how themes work: previously there was master `styles.css` and then any theme would load its css on top of it. that could be gradio default, 3rd party themes or actual built-in themes. they all used master `styles.css` which did most of heavy lifting. but making it be compatibile with every possibility on the planet is not viable.

so it was broken down: now there is `base.css` which is used if you're NOT using sdnext built-in themes which does very minimal work and then 3rd party theme is loaded on top of it. or there is `sdnext.css` which is loaded as base for all sdnext built-in themes that then can be much lighter and easier to maintain (that's why we have 6 built-in native themes now).

**Reply by Rex109** [reply]
Alright got it, thank you!

**Reply by Tillerz** [reply]
Btw, where are those colors defined, there's like --neutral-700 and stuff, and I couldn't find it anywhere.

**Reply by vladmandic** [reply]
> Btw, where are those colors defined, there's like --neutral-700 and stuff, and I couldn't find it anywhere.

For Sdnext native themes, it's in actual themes.
For Gradio or 3rd party themes, itd inside Gradio itself. Can be overriden in user.css.

**Comment by PresTrembleyIIIEsq** [comment]
Sorry to resurrect this months later, and this is probably just me not understanding at all, but I see the comment above about base.css vs. sdnext.css. When I launch SDNext, I see the following:

`INFO     Load UI theme: name="orchid-dreams" style=Dark base=sdnext.css`

Which sounds to me like it's using the newer version (and I even entirely removed the base.css file as a quick test). But I still see the same issue shown in the screenshots in the original post above.

I tried the orchid-dreams theme because it looked newer, but this also happens which each theme I've tried so far. Are there specific ones that work better with sdnext.css, or is there anything else I need to do to make sure I'm using the latest and most supported themes?

**Comment by vladmandic** [comment]
That theme should work - post a screenshot.

**Comment by PresTrembleyIIIEsq** [comment]
Additional info: This is in updated Firefox, no browser extensions, new re-install of SD.Next, running on Manjaro. Some field/buttons work, but most of the important ones like Generate don't do anything at all.

![Screenshot](https://github.com/vladmandic/automatic/assets/140981996/c03df85b-1d76-4057-b452-e75468361b4f)

**Reply by vladmandic** [reply]
This thread is about UI look/feel, if buttons don't work, please document and create an issue.
